### PR TITLE
add Go edition variants

### PR DESCRIPTION
**Pull request by tenzap**
Comments are welcome.
To add packages, new google certificates need to be added to the opengapps keystore #949 

Go variants are suffixed with "_go" (except super which contains in itself the go packages).

For example 
`bash build_gapps.sh arm 31 pico_go`

### CODE REVIEW

**Comment by mfonville** [comment]
Thanks, sorry that I didn't take a look earlier yet. But I will need some time to review

**Review by mfonville — COMMENTED** [review]
Thanks for your big PR to set-up the Go-packages.

Did you do extensive testing on these changes? It are quite a lot of changes, so it is good to make sure

**Comment by tenzap** [comment]
> Did you do extensive testing on these changes? It are quite a lot of changes, so it is good to make sure

I don't know what you mean by extended, but I tested it with lineageos & google system images using emulated device with [this script](https://gist.github.com/tenzap/253c8918ee488c3874f20a563f458510) that works from android 8 up to android 12.
I also tested with different variants.

Tests also led to filing #951 & #952

File: scripts/build_gapps.sh [hunk 1]
```diff
@@ -96,6 +97,11 @@ if [ -z "$SUPPORTEDVARIANTS" ]; then
   echo "ERROR: Unknown variant! aborting..."
   exit 1
 fi
+if [[ "$VARIANT" =~ _go$ ]] && [ "$API" -lt "27" ]; then
```
[the remark this answers is not in the document]
Comment (tenzap) [reply scripts/build_gapps.sh:100]: Probably. What would be the correct way to check this (ends with `_go`) if not in bash? I couldn't find an alternative in a quick search.

[the remark this answers is not in the document]
Comment (mfonville) [reply scripts/build_gapps.sh:100]: Sorry for my slow respons.
In these situation you can use a `case` statement. I think you already used somewhere a matching against `*_go)` in a case-statement somewhere

**Comment by mfonville** [comment]
Thanks for all the testing, if you can fix that bashism I will merge it~

File: scripts/build_gapps.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (tenzap) [reply scripts/build_gapps.sh]: thanks for the tip. Done.

**Comment by mfonville** [comment]
Could you rebase on current master to resolve the conflict?

**Comment by tenzap** [comment]
I don't see any conflict. I solved it in 6c6b593

**Comment by tenzap** [comment]
After merge, you may also consider #949

**Comment by mfonville** [comment]
Github still mentions the conflict, can you rebase on https://github.com/opengapps/opengapps/commit/897620fcca5c0bada944eaec73ffd3d59aff280a ?

**Comment by tenzap** [comment]
IMG_20220611_192553
That's what I have

**Comment by mfonville** [comment]
That is so weird, I have "This branch cannot be rebased due to conflicts".
I will rebase manually

**Comment by tenzap** [comment]
Ok. Thanks again for merging.

**Pull request merged**